otelcol/command.go

```diff
@@ -56,6 +71,9 @@ func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet) error
 	// TODO: Remove this after CollectorSettings.ConfigProvider is removed and instead
 	// do it in the builder.
 	if len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
+		if enforceProviders {
+			return errors.New("at least one Provider must be supplied")
+		}
 		set.ConfigProviderSettings = newDefaultConfigProviderSettings(resolverSet.URIs)
 	}
```

Comment:
```suggestion
	if !enforceProviders && len(resolverSet.ProviderFactories) == 0 && len(resolverSet.ConverterFactories) == 0 {
		set.ConfigProviderSettings = newDefaultConfigProviderSettings(resolverSet.URIs)
	}
```
I just realized that we can essentially make this if statement into a no-op and let the downstream packages handle it. Not a huge deal since hitting this error means that a distro was built incorrectly and the error won't be of much help to end users.

Reply:
I'd rather return the error, since we know here something is not right.  I see it as `otelcol.NewCommand` enforcing behavior it wants instead of depending on other packages, which could technically change, enforcing it.

Reply:
That's okay with me, it's functionally equivalent either way and we'll remove this soon. Let's leave it as-is.